crates/router/src/connector/authorizedotnet/transformers.rs

```diff
@@ -68,31 +68,86 @@ enum PaymentDetails {
     BankRedirect,
 }
 
-impl TryFrom<api_models::payments::PaymentMethodData> for PaymentDetails {
-    type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(value: api_models::payments::PaymentMethodData) -> Result<Self, Self::Error> {
-        match value {
+fn get_pm_and_subsequent_auth_detail(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> Result<
+    (
+        PaymentDetails,
+        Option<ProcessingOptions>,
+        Option<SubsequentAuthInformation>,
+    ),
+    error_stack::Report<errors::ConnectorError>,
+> {
+    match item
+        .request
+        .mandate_id
+        .to_owned()
+        .and_then(|mandate_ids| mandate_ids.mandate_reference_id)
+    {
+        Some(api_models::payments::MandateReferenceId::NetworkMandateId(
+            original_network_trans_id,
+        )) => {
+            let processing_options = Some(ProcessingOptions {
+                is_subsequent_auth: true,
+            });
+            let subseuent_auth_info = Some(SubsequentAuthInformation {
+                original_network_trans_id,
+                reason: Reason::Resubmission,
+            });
+            match item.request.payment_method_data {
+                api::PaymentMethodData::Card(ref ccard) => {
+                    let payment_details = PaymentDetails::CreditCard(CreditCardDetails {
+                        card_number: ccard.card_number.clone(),
+                        // expiration_date: format!("{expiry_year}-{expiry_month}").into(),
```

Comment:
remove commented code

Reply:
sure, it was there before maybe for some reference